### PR TITLE
Student t test

### LINKED ISSUE #6: Student's t-test

### PR DESCRIPTION
**Pull request by jacked97**

### CODE REVIEW

**Review by nikitml — CHANGES_REQUESTED** [review]
Hey, so, in general, there's a structure and a part of the topic covered, however, I'd propose to fix these small issues:
1. Explain the assumptions/prerequesites/corener stones about the data you're using
2. I think the interactivity of the charts could address more of the topic and not only the template version of `@interact`
3. Also, I think, you could split the code into smaller parts with a reasoning before the cell and the results of the execution after the cell
4. In the _Conclusion_ part I'd love to have more of a digest/tl;dr version of the whole article, but w/o code. Just, maybe, to refresh the definition (that you have in the first part), remind of the caveats/cornerstones and applicability of the test and so on

Hope you will find no difficulties fixing these issues and will have you work improved dramatically, so the next year students would have a great material to study! Thanks!

File: Student-T-Test/Student's t-test.ipynb [hunk 1]
```diff
@@ -0,0 +1,869 @@
...
+      "Requirement already satisfied: six in /Users/abdullah/anaconda3/lib/python3.7/site-packages (from chart_studio) (1.12.0)\n",
+      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Users/abdullah/anaconda3/lib/python3.7/site-packages (from requests->chart_studio) (1.24.2)\n",
+      "Requirement already satisfied: idna<2.9,>=2.5 in /Users/abdullah/anaconda3/lib/python3.7/site-packages (from requests->chart_studio) (2.8)\n",
+      "Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /Users/abdullah/anaconda3/lib/python3.7/site-packages (from requests->chart_studio) (3.0.4)\n",
+      "Requirement already satisfied: certifi>=2017.4.17 in /Users/abdullah/anaconda3/lib/python3.7/site-packages (from requests->chart_studio) (2019.9.11)\n"
+     ]
+    }
+   ],
+   "source": [
+    "! pip install ipywidgets\n",
```
Comment (apoyezzhayev) [inline Student-T-Test/Student's t-test.ipynb]: Split install and import to two different cells for convenient repeated  launch of the notebook.

File: Student-T-Test/Student's t-test.ipynb [hunk 2]
```diff
@@ -0,0 +1,869 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "jNm4a9nY23l2"
+   },
+   "source": [
+    "`interact` [documentation](https://ipywidgets.readthedocs.io/en/stable/examples/Using%20Interact.html)"
```
Comment (apoyezzhayev) [inline Student-T-Test/Student's t-test.ipynb]: Why do we need it, it's pretty intuitive

File: Student-T-Test/Student's t-test.ipynb [hunk 3]
```diff
@@ -0,0 +1,869 @@
...
+      "Liver Patients: Mean: 0.9141787439613527, Variance: 0.10637547402650555, Size: 416\n",
+      "Non-liver Patients: Mean: 1.0295757575757576, Variance: 0.08251384331116045, Size: 167\n"
+     ]
+    }
+   ],
+   "source": [
+    "lp_mean = liver_patient.mean()\n",
+    "nlp_mean = non_liver_patient.mean();\n",
+    "\n",
+    "lp_variance = liver_patient.var()\n",
```
Comment (apoyezzhayev) [inline Student-T-Test/Student's t-test.ipynb]: Which kind of variance it calculates (biased or not)? Make a remark on it.

**Review by apoyezzhayev — COMMENTED** [review]
I really liked the section with Pseudo Understanding Of t-Test but the notebook is quite laconic.  
1) Add some visualizations:
 - t-distribution with colored tails
 - t-distribution vs normal distribution for different N
2) Add information about connection between Normal distribution and t-distribution as well as between one-sample t-test and z-test
3) Add some structure for t-test with examples:
  - Paired / Unpaired
  - One sample / two samples location tests
4) Quite interesting intro but is optional (IMHO)

File: Student-T-Test/Student's t-test.ipynb [hunk 4]
```diff
@@ -0,0 +1,869 @@
...
+    "- Student t-Test dependent samples\n",
+    "- Student t-Test independent samples\n",
+    "- Application with real world dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pseduo Understanding Of t-Test\n",
```
Comment (apoyezzhayev) [inline Student-T-Test/Student's t-test.ipynb]: typo: Pseudo

**Comment by nikitml** [comment]
@jacked97 please check the comments, there's "action required"
@ZamiraKholmatova please review

**Comment by jacked97** [comment]
@lozhn & @apoyezzhayev Thanks for the feedback!
@lozhn I have pushed the latest changes based on your review. 
@apoyezzhayev Few things I skipped from your review, since there is another notebook with T-Distribution, so I tried to keep it specific to T-Test.

**Review by jacked97 — COMMENTED** [review]
Completed in the latest commit

**Review by ZamiraKholmatova — COMMENTED** [review]
1. Very large intro (it is only my opinion, you can keep it if you like);
2. Some visualizations of distribution will be good (add them => your work will be excellent).

**Review by nikitml — APPROVED** [review]
TOP 👍

**Review by apoyezzhayev — APPROVED** [review]
Good!

**Review by ZamiraKholmatova — APPROVED** [review]

**Pull request merged**